### PR TITLE
SCT-1221 Add user handler set class with cacheing

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/jgikbase/idmapping/core/user_handler.py [hunk 1]
```diff
@@ -44,12 +48,96 @@ def is_valid_user(self, username: Username) -> Tuple[bool, Optional[int], Option
...
+        :param cache_timer: the timer used for cache expiration. Defaults to time.time.
+        :param cache_max_size: the maximum size of the token -> user and username -> validity
+            caches.
+        :param cache_user_expiration: the default expiration time for the token -> user cache in
+            seconds. This time can be overridden by a user handler on a per token basis.
+        :param cache_is_valid_expiration: the default expiration time for the  username ->
+            validity cache. This time can be overridden by a user handler on a per user basis.
+        """
+        no_Nones_in_iterable(user_handlers, 'user_handlers')
+        self._handlers = {handler.get_authsource_id(): handler for handler in user_handlers}
```
Comment (jayrbolton) [inline src/jgikbase/idmapping/core/user_handler.py]: maybe there is a better word than "handler" that is more descriptive? That word doesn't convey meaning to me -- you could say all programs handle something.

File: src/jgikbase/idmapping/core/user_handler.py [hunk 2]
```diff
@@ -2,9 +2,12 @@
 from jgikbase.idmapping.core.tokens import Token
 from jgikbase.idmapping.core.user import User, AuthsourceID, Username
 from jgikbase.idmapping.storage.id_mapping_storage import IDMappingStorage
-from jgikbase.idmapping.core.util import not_none
+from jgikbase.idmapping.core.util import not_none, no_Nones_in_iterable
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/core/user_handler.py]: changed to arg_check

File: src/jgikbase/idmapping/core/user_handler.py (hunk 1 above)
Comment (MrCreosote) [reply src/jgikbase/idmapping/core/user_handler.py]: Changed handler to lookup all over the code base, so e.g. LocalUserHandler is now LocalUserLookup.

**Comment by MrCreosote** [comment]
Note the changes described are in my local branch; they'll be coming in later PRs.

**Pull request merged**